content/things/rust-to-webassembly.md

```diff
@@ -0,0 +1,637 @@
+
+Now for the actual implementation of the `GlobalAlloc` trait:
+
+```rust
+unsafe impl GlobalAlloc for SimpleAllocator {
+    unsafe fn alloc(&self, layout: Layout) -> *mut u8 {
+        let size = layout.size();
+        let align = layout.align();
+
+        // Find the next address that has the right alignment.
+        let idx = (*self.head.get()).next_multiple_of(align);
+        // Bump the head to the next free byte
+        *self.head.get() = idx + size;
+        let arena: &mut [u8; ARENA_SIZE] = &mut (*self.arena.get());
+        // If we ran out of arena space, we return a null pointer, which
+        // signals a failed allocation.
+        match arena.get_mut(idx) {
+            Some(item) => item as *mut u8,
+            _ => core::ptr::null_mut(),
+        }
+    }
+
+    unsafe fn dealloc(&self, _ptr: *mut u8, _layout: Layout) {
+        /* lol */
+    }
+}
+```
+
+As with any good bump allocator, you can’t free memory.
+
+`#[global_allocator]` is not limited to `#[no_std]`! You can override Rust's default allocator and replace it with your own, as Rust's default allocator consumes about 10K of Wasm space.
+
+### wee_alloc & lol_alloc
+
+You don’t have to implement the allocator yourself, of course. In fact, it’s probably advisable to rely on a well-tested implementation. Dealing with bugs in the allocator and subtle memory corruption is not fun. 
+
+Many guides recommend [`wee_alloc`][wee_alloc], which is a very small (<1KB) allocator written by the Rust WebAssembly team that can also free memory. Sadly, it seems unmaintained and has an open issue about memory corruption and leaking memory. There’s a [`lol_alloc`][lol_alloc] which seems to aspire to replace `wee_alloc` and provides a modular setup so you can tune the allocator to your needs: 
+
+```rust
+use lol_alloc::{FreeListAllocator, LeakingAllocator};
+use lol_alloc::{LockedAllocator, AssumeSingleThreaded};
+
+// Better version of our bump allocator above.
+#[global_allocator]
+static ALLOCATOR: AssumeSingleThreaded<LeakingAllocator> = 
+  AssumeSingleThreaded::new(LeakingAllocator::new());
+
+// "Proper" allocator with added thread-safety
+#[global_allocator]
+static ALLOCATOR: LockedAllocator<FreeListAllocator> = 
+  LockedAllocator::new(FreeListAllocator::new());
+
+// ...
+```
+
+I don’t have any first-hand experience with `lol_alloc` outside of writing this article, but it looks good!
+
+## wasm-bindgen
+
+Now that we've done pretty much everything the hard way, we have earned a look at the easy way of writing Rust for WebAssembly, which is using [wasm-bindgen].
```

Comment:
I tend to avoid words like `easy` or `this is simple` because they might not be universally true and can make a beginner feel a bit inadequate (anxiety anxiety anxiety :)). 
Something like "out of the box," or "straightforward," or "standard" way could sound better. 

It's a style preference, but perhaps something to perhaps consider for future blog posts.